ci_framework/roles/edpm_prepare/tasks/main.yml

```diff
@@ -0,0 +1,100 @@
+- name: Prepare storage in CRC
+  when:
+    - cifmw_use_crc is defined
+    - cifmw_use_crc | bool
+  ansible.builtin.include_role:
+    name: 'install_yamls_makes'
+    tasks_from: 'make_crc_storage'
+
+- name: Create install_yamls output directory
+  ansible.builtin.file:
+    path: "{{ cifmw_edpm_prepare_intall_yamls_output_dir }}"
+    state: directory
+
+- name: Set install_yamls Makefile environment variables
+  ansible.builtin.set_fact:
+    cifmw_edpm_prepare_common_env: |
+      {% if cifmw_edpm_prepare_password is defined %}
+      PASSWORD: {{ cifmw_edpm_prepare_password }}
+      {% endif %}
+      {% if cifmw_edpm_prepare_secret is defined %}
+      SECRET: {{ cifmw_edpm_prepare_secret }}
+      {% endif %}
+      OUT: {{ cifmw_edpm_prepare_intall_yamls_output_dir }}
+    cifmw_edpm_prepare_make_openstack_env: |
+      {% if cifmw_operator_build_meta_name in cifmw_operator_build_output.operators %}
+      OPENSTACK_IMG: {{ cifmw_operator_build_output.operators[cifmw_operator_build_meta_name].image }}
+      {% endif %}
+    cifmw_edpm_prepare_make_openstack_deploy_env: |
+      {% if cifmw_operator_build_meta_name in cifmw_operator_build_output.operators %}
+      OPENSTACK_BRANCH: ""
+      GIT_CLONE_OPTS: "-l"
+      OPENSTACK_REPO: cifmw_operator_build_output.operators[cifmw_operator_build_meta_name].git_src_dir
+      {% endif %}
+
+- name: Prepare inputs
+  vars:
+    make_input_env: "{{ cifmw_edpm_prepare_common_env | from_yaml }}"
+  ansible.builtin.include_role:
+    name: 'install_yamls_makes'
+    tasks_from: 'make_input'
+
+- name: Install OpenStack operator
+  vars:
+    make_openstack_env: "{{ cifmw_edpm_prepare_common_env | from_yaml |
+      combine(cifmw_edpm_prepare_make_openstack_env | from_yaml)}}"
+  ansible.builtin.include_role:
+    name: 'install_yamls_makes'
+    tasks_from: 'make_openstack'
+
+- name: Wait for OpenStack subscription creation
+  ansible.builtin.shell: |
+    oc get sub openstack-operator --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }} > /dev/null 2>&1 &&
+      oc get sub openstack-operator --namespace={{ cifmw_install_yamls_defaults['NAMESPACE'] }} -o=jsonpath='{.status.installplan.name}'
+  until: cifmw_edpm_prepare_wait_installplan_out.rc == 0 and cifmw_edpm_prepare_wait_installplan_out.stdout != ""
+  register: cifmw_edpm_prepare_wait_installplan_out
+  retries: 30
+  delay: 5
```

Comment:
or can we move these tasks to a seperate playbook and invoke based on some condition like install_openstack_operator | default ('true') | bool? This way I can define one var in ci and job is done.

Reply:
So, you need to choose what operators to deploy? Or is the openstack metaoperator the only one to skip? I can create a variable with a default for that

Reply:
We want to skip the openstack metaoperator installation.

Reply:
I'll add the cifmw_edpm_skip_openstack_operator with a default value of false. Thanks for the point!

Reply:
It can be, but, that's not that easy, as we may need to tune the OpenStackControlPlane CR, is doable and it should be somehting I'd like to see, but that will take some more time. Skipping the metaoperator is an easy task.